packages/react/use-previous/src/usePrevious.tsx

```diff
@@ -3,15 +3,18 @@ import * as React from 'react';
 function usePrevious<T>(value: T) {
   // The ref object is a generic container whose current property is mutable ...
   // ... and can hold any value, similar to an instance property on a class
-  const ref = React.useRef<T>(value);
+  const ref = React.useRef({ value, previous: value });
```

Comment:
Just thinking, `previous` should probably be initially `undefined`? the original hook doesn't account for that but I think that makes more sense?

Reply:
We purposefully made it default to `value` so that hooks and things didn't run on initial render. The hook is usually used to signal a "value change" and on initial render, the value hasn't changed.